ui/src/ReactCustomHooks/useWebSocket/useWebSocket.hook.js

```diff
@@ -0,0 +1,113 @@
+import { useState, useEffect } from 'react';
+import { NO_OP, EMPTY_OBJ } from 'Utils';
+
+const WEBSOCKET = {
+  CONNECTING: 0,
+  OPEN: 1,
+  CLOSING: 2,
+  CLOSED: 3,
+  INVALID: 'Invalid WebSocket function provided. See documentation',
+  SENT: true,
+  NOT_SENT: false,
+};
+
+const SOCKET_NOT_OPEN_HANDLER = () => {
+  // eslint-disable-next-line no-console
+  console.error('Socket is not open');
+  return WEBSOCKET.NOT_SENT;
+};
+
+const DEFAULT_STATE = {
+  send: SOCKET_NOT_OPEN_HANDLER,
+  currentState: WEBSOCKET.CLOSED,
+};
+
+const openWebSocket = (
+  getWebsocket,
+  { onOpen = NO_OP, onMessage = NO_OP, onError, onClose = NO_OP },
+  socketState,
+  updateSocketState
+) => {
+  let closeSocket = NO_OP;
+
+  if (
```

Comment:
No url support?

Reply:
Change in API since we last discussed it - expects a callback to return a websocket. URL is one of many options which could be provided - felt more flexible/intuitive to have one expected input here